decisions/log/0026-async-request-by-default.md

```diff
@@ -0,0 +1,178 @@
+# 26. Switch to async request/response as mandatory (default) behaviour.
+
+Date: 2023-11-07
+
+## Status
+
+Proposed
+
+## Context
+
+The current version of the specification[^1] favors synchronous communication over asynchronous communication. As, especially in the early stages of the network, most requests will target not yet existing PF values, the existing approach using synchronous `ListFootprints` calls will not work. PCF data is often not available and is calculated after the request for a specific product arrives at the data provider (Supplier). 
+
+Furthermore managing the access rights upfront for `ListFootprints` requests are complex for large enterprises with different Business Units and many products. Routing requests on specific products is easier than providing PCF data and managing access rights upfront. 
+
+The seconds synchronous request is `GetFootprint` which uses PF IDs to select the requested PCF. But _PF IDs_ are a detail which is usually unknown to the data requester (customer). Data requesters usually only know _Material Identifiers_ (e.g. catalog number, part number).
+Product specific requests would in addition enable data providers (suppliers) to prioritize the calculation of PCF when these are not available yet.
+
+The specification[^1] already adresses all this by supporting `PF Request Events and Responses`. But this feature is _optional_ at the moment whereas synchronous communication mechanisms (as `ListFootprints` or `GetFootprint`) are defined as _mandatory_. This complicates or even block integration of existing PCF exchange apps as those apps usually work in an asynchrounous and productId focused way.
+
+Other exchange networks like Catena-X also use this asynchronous/material approach. With the current specification the PACT API requires the user to handle two APIs with different logic. 
+
+## Decision
+
+The following changes to the tech spec Bikeshed file[^1] shall be made:
+
+1. Change content of `Host System Minimum Requirements` section:
+   * Change `Action ListFootprints` from _mandatory_ to _optional_, i.e. move it from _MUST_ to _SHOULD_ paragraph. 
+   * Change `Action Events` from _optional_ to _mandatory_, i.e. move it from _SHOULD__ to _MUST_ paragraph. 
```

Comment:
while I understand the rationale for there is need for an upfront setup to respond to a PCF request and hence making this Action Events Mandatory vs the ListFootPrints endpoint, but this will definitely a breaking change and also cause a lot of re-work to for the SPs who have already implemented the earlier versions of the Tech Spec. How about if we keep both these as-is and depending on the availability of the necessary logic/data ready in the ListFootprint endpoint implementation, Solutions can either respond with the data (if available), otherwise return a valid HTTP Error response with an appropriate status code (e..g 202) that the data is not available but it will be available in future?

Reply:
@prasadt1 you are right, breaking changes always hurt. A corresponding HTTP code could be a smooth solution here! Just wondering wich one would be a good choice from a semantic point of view:
- _202 Accepted_ will intend that the caller would just have to wait, which is not true - the caller would have to ask via async call and include a productId to get results.
- _303 See Other_ may be a choice as it tell the client to go to a different URL. Only challange here: intention is that you can do a GET on the endpoint given. But as this is a SHOULD in the HTTP standard, it may be a way to go?

Reply:
Changed ADR to keep ListFootprints as mandatory and introduce a new status code (HTTP-303) instead.

Reply:
Assuming we're following semver conventions - so this would also be a major version change too @prasadt1 (i.e.: transitioning api endpoints from optional to mandatory)